apimodels/agent_models.go

```diff
@@ -179,9 +203,39 @@ func (ch *CreateHost) Validate() error {
 	if ch.TeardownTimeoutSecs < 60 || ch.TeardownTimeoutSecs > 604800 {
 		catcher.Add(errors.New("timeout_teardown_secs must be between 60 and 604800"))
 	}
+
 	return catcher.Resolve()
 }
 
+func (ch *CreateHost) setNumHosts() error {
+	if ch.NumHosts == "" {
+		ch.NumHosts = "1"
+	} else {
+		numHosts, err := strconv.Atoi(ch.NumHosts)
+		if err != nil {
+			return errors.Errorf("problem parsing '%s' as an int", ch.NumHosts)
+		}
+		if numHosts > 10 || numHosts < 0 {
+			return errors.New("num_hosts must be between 1 and 10")
+		} else if numHosts == 0 {
+			ch.NumHosts = "1"
+		}
+	}
+	return nil
+}
+
+func (ch *CreateHost) Validate() error {
+	if ch.CloudProvider == ProviderEC2 || ch.CloudProvider == "" { //default
+		return ch.ValidateEC2()
+	}
+
+	if ch.CloudProvider == ProviderDocker {
+		return ch.ValidateDocker()
+	}
+
+	return errors.New("only ec2 and docker are currently supported cloud providers")
```

Comment:
Use the constants: `return errors.Errorf("Cloud provider must be either '%s' or '%s'", ProviderEC2, ProviderDocker)`